lib/core/request.js

```diff
@@ -97,7 +98,13 @@ class Request {
 
     this.upgrade = upgrade || null
 
-    this.path = path
+    if (params && Object.keys(params).length > 0) {
+      const searchParams = new URLSearchParams(params)
+      const delimiter = path.indexOf('?') === -1 ? '?' : '&'
```

Comment:
What if path just ends with ? without any params?

Reply:
Tried `https://openlibrary.org/api/books?&bibkeys=ISBN:0201558025` and `https://fakestoreapi.com/products?&limit=3`, it still works as expected, so it seems like it's a generally acceptable input. Axios is not protecting against this case either.

Reply:
`path.indexOf('?') === -1` -> `path.includes('?')`

Reply:
isn't indexOf faster?

Reply:
I don't see why it should be faster, but anyway it's unlikely to matter compared to the rest of the work done here.

Reply:
Did the change, but I can share a benchmark later, there is a difference, but yeah, should be negligible in the context.